Added access control info to the vcd vdc info <vdc_name> command. and vcd vapp info <vapp_name>

anusuyar opened the pull request:
```
Added access control info to the vcd vdc info <vdc_name> command. and vcd vapp info <vapp_name>

By default, the VDC will have the following control access settings (Case1)
<ControlAccessParams>
    <IsSharedToEveryone>true</IsSharedToEveryone>
    <EveryoneAccessLevel>ReadOnly</EveryoneAccessLevel>
</ControlAccessParams>

But we can update it to have user specific access control (Case2)
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<ControlAccessParams>
   <IsSharedToEveryone>false</IsSharedToEveryone>
   <AccessSettings>
      <AccessSetting>
        <Subject
            type="application/vnd.vmware.admin.user+xml"
            href="https://bos1-vcd-sp-static-199-204.eng.vmware.com/api/admin/user/d69620d1-1be4-41bc-8db3-ef1006dac028"/>
         <AccessLevel>ReadOnly</AccessLevel>
      </AccessSetting>
    </AccessSettings>
</ControlAccessParams>

Command output for Case1, 
C:\sources\github\service_ecosystem\vcd-cli (development) 
(vcd-cli) λ vcd vdc info vdc2(everyone_access_level and is_shared_to_everyone are the new access specific params)
property               value
---------------------  -------------------------------------------------------------------------------------------------------
everyone_access_level   ReadOnly
allocation_model       AllocationPool
cpu_capacity           {'units': 'MHz', 'allocated': '1980', 'limit': '1980', 'reserved': '990', 'used': '0', 'overhead': '0'}
id                     a9fa0a54-1c9d-44a6-81e3-4deaad599265
in_use                 False
is_enabled             True
is_shared_to_everyone  True
mem_capacity           {'units': 'MB', 'allocated': '1187', 'limit': '1187', 'reserved': '593', 'used': '0', 'overhead': '0'}
name                   vdc2
org                    AnuOrg
supported_hw           ['vmx-04', 'vmx-07', 'vmx-08', 'vmx-09', 'vmx-10', 'vmx-11']
vm_quota               100

Command output for Case2 (everyone_access_level param is absent in this case, but instead we have new per user access control settings.
C:\sources\github\service_ecosystem\vcd-cli (development) 
(vcd-cli) λ vcd vdc info vdc1
property               value
---------------------  --------------------------------------------------------------------------------------------------------
allocation_model       AllocationPool
cpu_capacity           {'units': 'MHz', 'allocated': '2200', 'limit': '2200', 'reserved': '1100', 'used': '0', 'overhead': '0'}
id                     eee20b45-8c2f-4b9a-bdd7-fc9cb224045a
in_use                 True
is_enabled             True
is_shared_to_everyone  False
mem_capacity           {'units': 'MB', 'allocated': '1310', 'limit': '1310', 'reserved': '655', 'used': '0', 'overhead': '0'}
name                   vdc1
org                    AnuOrg
supported_hw           ['vmx-04', 'vmx-07', 'vmx-08', 'vmx-09', 'vmx-10', 'vmx-11']
vm_quota               100

Access control settings:

access_level    user_href                                                                                              user_name
--------------  -----------------------------------------------------------------------------------------------------  -----------
ReadOnly        https://bos1-vcd-sp-static-199-204.eng.vmware.com/api/admin/user/d69620d1-1be4-41bc-8db3-ef1006dac028  user

Same applies for vapp, except EveryoneAccessLevel is absent in the XML.

```

Commit:
Merge branch 'master' into development

Commit:
Removed org_name in the update_org.
Since we are going to create a new method reload org. Not updating self.resource after updating
the org.

Commit:
Merge branch 'master' into development

Commit:
Merge branch 'master' into development

Commit:
Added access control info to the vcd vdc info <vdc_name> command.

By default, the VDC will have the following control access settings (Case1)
<ControlAccessParams>
    <IsSharedToEveryone>true</IsSharedToEveryone>
    <EveryoneAccessLevel>ReadOnly</EveryoneAccessLevel>
</ControlAccessParams>

But we can update it to have user specific access control (Case2)
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<ControlAccessParams>
   <IsSharedToEveryone>false</IsSharedToEveryone>
   <AccessSettings>
      <AccessSetting>
        <Subject
            type="application/vnd.vmware.admin.user+xml"
            href="https://bos1-vcd-sp-static-199-204.eng.vmware.com/api/admin/user/d69620d1-1be4-41bc-8db3-ef1006dac028"/>
         <AccessLevel>ReadOnly</AccessLevel>
      </AccessSetting>
    </AccessSettings>
</ControlAccessParams>

Command output for Case1,
C:\sources\github\service_ecosystem\vcd-cli (development)
(vcd-cli) λ vcd vdc info vdc2(everyone_access_level and is_shared_to_everyone are the new access specific params)
property               value
---------------------  -------------------------------------------------------------------------------------------------------
everyone_access_level   ReadOnly
allocation_model       AllocationPool
cpu_capacity           {'units': 'MHz', 'allocated': '1980', 'limit': '1980', 'reserved': '990', 'used': '0', 'overhead': '0'}
id                     a9fa0a54-1c9d-44a6-81e3-4deaad599265
in_use                 False
is_enabled             True
is_shared_to_everyone  True
mem_capacity           {'units': 'MB', 'allocated': '1187', 'limit': '1187', 'reserved': '593', 'used': '0', 'overhead': '0'}
name                   vdc2
org                    AnuOrg
supported_hw           ['vmx-04', 'vmx-07', 'vmx-08', 'vmx-09', 'vmx-10', 'vmx-11']
vm_quota               100

Command output for Case2 (everyone_access_level param is absent in this case, but instead we have new per user access control settings.
C:\sources\github\service_ecosystem\vcd-cli (development)
(vcd-cli) λ vcd vdc info vdc1
property               value
---------------------  --------------------------------------------------------------------------------------------------------
allocation_model       AllocationPool
cpu_capacity           {'units': 'MHz', 'allocated': '2200', 'limit': '2200', 'reserved': '1100', 'used': '0', 'overhead': '0'}
id                     eee20b45-8c2f-4b9a-bdd7-fc9cb224045a
in_use                 True
is_enabled             True
is_shared_to_everyone  False
mem_capacity           {'units': 'MB', 'allocated': '1310', 'limit': '1310', 'reserved': '655', 'used': '0', 'overhead': '0'}
name                   vdc1
org                    AnuOrg
supported_hw           ['vmx-04', 'vmx-07', 'vmx-08', 'vmx-09', 'vmx-10', 'vmx-11']
vm_quota               100

Access control settings:

access_level    user_href                                                                                              user_name
--------------  -----------------------------------------------------------------------------------------------------  -----------
ReadOnly        https://bos1-vcd-sp-static-199-204.eng.vmware.com/api/admin/user/d69620d1-1be4-41bc-8db3-ef1006dac028  user

anusuyar commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/anusuyar/pyvcloud/8)
<!-- Reviewable:end -->

Commit:
Retrieve vapp access control

sahithi commented:
Reviewed 1 of 3 files at r1, 4 of 4 files at r2.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[pyvcloud/vcd/utils.py, line 259 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0QrR7tBJeWGlFBstTG:-L0QrR7uh22Hr5AMG34Q:b-s3tagb) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/utils.py#L259)):*
> ```Python
> 
> 
> def control_access_params_to_dict(control_access):
> ```

method name : access_control_params_to_dict sounds better?
parameter name: access_control sounds better?

---

*[pyvcloud/vcd/utils.py, line 269 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0Qrh33yxOQbuPskO6G:-L0Qrh34AjuBhzswMcqM:bk8e51a) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/utils.py#L269)):*
> ```Python
> 
> 
> def process_access_control_setting(control_access):
> ```

method name: Basically this is converting access_control to dict, please name it appropriately to indicate it is converting to dict. Then I suggest merge the other method in here (with added comments) . Two methods seem redundant to me.
param name : access_control

---

*[pyvcloud/vcd/utils.py, line 269 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0Qs9rwJ6gaZvvxCrEI:-L0Qs9rx-moyQ7W67hLP:b-5l640d) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/utils.py#L269)):*
> ```Python
> 
> 
> def process_access_control_setting(control_access):
> ```

None check on control_access?

---

*[pyvcloud/vcd/utils.py, line 277 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0QrsVfFJaU8ECe7P0S:-L0QrsVfFJaU8ECe7P0T:b1z94xk) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/utils.py#L277)):*
> ```Python
>                 control_access.AccessSettings.AccessSetting):
>             access_settings.append(access_settings_to_dict(access_setting))
>     result = control_access_params_to_dict(control_access)
> ```

result = control_access_params_to_dict.. can be moved to first line.

---

*[pyvcloud/vcd/vapp.py, line 278 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0QtcO-GLd90jOsffoz:-L0QtcO0uYhprsyrtdMS:bpsgozl) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/vapp.py#L278)):*
> ```Python
>         """
>         Get the access control settings of the vapp.
>         :return: Access control settings of the vapp.
> ```

return dict

---

*[pyvcloud/vcd/vapp.py, line 285 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0Qtf9rJVWMSmP7r9d3:-L0Qtf9rJVWMSmP7r9d4:beopiyn) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/vapp.py#L285)):*
> ```Python
>             self.resource, RelationType.DOWN,
>             EntityType.CONTROL_ACCESS_PARAMS.value)
>         return process_access_control_setting(control_access)
> ```

return access_control_settings_to_dict(access_control) makes more sense.

---

*[tests/vcd_vapp.py, line 265 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0Qtz1JY8dpJz2dTqq9:-L0Qtz1JY8dpJz2dTqqA:b-o4ifwa) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/tests/vcd_vapp.py#L265)):*
> ```Python
>         vapp = VApp(self.client, resource=vapp_resource)
>         access_control = vapp.get_access_control_settings()
>         assert len(access_control) > 0
> ```

assert access_control not None? It doesn't return list of records, right?

---

*[tests/vcd_vdc.py, line 29 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0QuEkSslqtjjEw-dIj:-L0QuEkSslqtjjEw-dIk:b-o1jpvz) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/tests/vcd_vdc.py#L29)):*
> ```Python
>         vdc = VDC(self.client, resource=vdc_resource)
>         access_control = vdc.get_access_control_settings()
>         assert len(access_control) > 0
> ```

assert access_control not None?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8)*
<!-- Sent from Reviewable.io -->

rocknes commented:
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[pyvcloud/vcd/utils.py, line 249 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0RKffCefsDdOYvpNum:-L0RKffCefsDdOYvpNun:b-y31hk9) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/utils.py#L249)):*
> ```Python
> 
> 
> def access_settings_to_dict(access_setting):
> ```

Remove the s from settings, we are converting one access_setting to dict

---

*[pyvcloud/vcd/utils.py, line 259 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0QrR7tBJeWGlFBstTG:-L0RHbIYQEI3c3GWe8Ff:b-q889xc) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/utils.py#L259)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

method name : access_control_params_to_dict sounds better?
parameter name: access_control sounds better?
</blockquote></details>

control_access_param is the correct name since we are actually converting a ControlAccessParam object to dictionary. AccessControlSetting is a different param.
Rename the param from "control_access" to "control_access_param_xml"/"control_access_param_blob"

---

*[pyvcloud/vcd/utils.py, line 269 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0Qrh33yxOQbuPskO6G:-L0RJVvSCb6uuoazAZ7B:b-yghpr7) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/utils.py#L269)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

method name: Basically this is converting access_control to dict, please name it appropriately to indicate it is converting to dict. Then I suggest merge the other method in here (with added comments) . Two methods seem redundant to me.
param name : access_control

</blockquote></details>

^+1 
method name should be access_control_settings_to_dict

However I don't think the two functions should be merged. Rather the conrtol_access_param dict should have a field ['access_control_settings'] which will contain the list of output of this function (for different access_control_setting).

And the caller i.e. info command can process the empty value as an indication to not show this data, or show the data in a different table.

---

*[pyvcloud/vcd/utils.py, line 271 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0RKA4ZhW_AjQOU0XAg:-L0RKA4ZhW_AjQOU0XAh:b-ze7z08) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/utils.py#L271)):*
> ```Python
> def process_access_control_setting(control_access):
>     access_settings = []
>     if hasattr(control_access, 'AccessSettings') and \
> ```

Change the param to access_control_settings

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[pyvcloud/vcd/utils.py, line 269 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0Qrh33yxOQbuPskO6G:-L0RXAO2py1exMKae4Im:b5gr1nk) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/utils.py#L269)):*
<details><summary><i>Previously, rocknes wrote…</i></summary><blockquote>

^+1 
method name should be access_control_settings_to_dict

However I don't think the two functions should be merged. Rather the conrtol_access_param dict should have a field ['access_control_settings'] which will contain the list of output of this function (for different access_control_setting).

And the caller i.e. info command can process the empty value as an indication to not show this data, or show the data in a different table.
</blockquote></details>

vcd-cli has a method `as_table` in the utils.py module that is a general utility method to display and array of entities, already transformed to dictionaries. `vapp_to_dict` is another example that flattens out a hierarchical object into key-value pairs

---

*[pyvcloud/vcd/vapp.py, line 285 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0Qtf9rJVWMSmP7r9d3:-L0RWFePa638A985qmDs:btghcbr) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/vapp.py#L285)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

return access_control_settings_to_dict(access_control) makes more sense.
</blockquote></details>

I have created ticket VCDA-335 to discuss a general approach to returning XML vs dict

---

*[tests/vcd_vdc.py, line 29 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0QuEkSslqtjjEw-dIj:-L0RYkElnMRw_rBFQQ0f:bng7b7d) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/tests/vcd_vdc.py#L29)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

assert access_control not None?
</blockquote></details>

this assert would fail if `access_control is None` anyways. Could it be `len(access_control) == 0` and still be a valid vDC configuration?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
Reviewed 1 of 3 files at r1, 4 of 4 files at r2.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8)*
<!-- Sent from Reviewable.io -->

Commit:
Addressing the review comment.
Displaying the acccess_control_settings as a flat list(as part of other info) instead of a table.

(vcd-cli) λ vcd vdc info vdc1
property                        value
------------------------------  --------------------------------------------------------------------------------------------------------
access-settings-1-access-level  ReadOnly
access-settings-1-subject-href  https://bos1-vcd-sp-static-199-204.eng.vmware.com/api/admin/user/d69620d1-1be4-41bc-8db3-ef1006dac028
access-settings-1-subject-name  user
access-settings-1-subject-type  application/vnd.vmware.admin.user+xml
allocation_model                AllocationPool
cpu_capacity                    {'units': 'MHz', 'allocated': '2200', 'limit': '2200', 'reserved': '1100', 'used': '0', 'overhead': '0'}
id                              eee20b45-8c2f-4b9a-bdd7-fc9cb224045a
in_use                          True
is_enabled                      True
is_shared_to_everyone           False
mem_capacity                    {'units': 'MB', 'allocated': '1310', 'limit': '1310', 'reserved': '655', 'used': '0', 'overhead': '0'}
name                            vdc1
org                             AnuOrg
supported_hw                    ['vmx-04', 'vmx-07', 'vmx-08', 'vmx-09', 'vmx-10', 'vmx-11']
vapp_templates                  []
vapps                           ['vapp2', 'vapp1']
vm_quota                        100

Also refactored get_catalog_access_settings to avoid duplication. and removed catalog control_access and move the info to catalog info command

anusuyar commented:
Review status: 0 of 6 files reviewed at latest revision, 10 unresolved discussions.

---

*[pyvcloud/vcd/utils.py, line 249 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0RKffCefsDdOYvpNum:-L0RXBxvspYPkqc0ELSi:b-896fix) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/utils.py#L249)):*
<details><summary><i>Previously, rocknes wrote…</i></summary><blockquote>

Remove the s from settings, we are converting one access_setting to dict
</blockquote></details>

Done.

---

*[pyvcloud/vcd/utils.py, line 259 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0QrR7tBJeWGlFBstTG:-L0RSKe_2WO5WpeLfOuZ:b-ulh4tp) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/utils.py#L259)):*
<details><summary><i>Previously, rocknes wrote…</i></summary><blockquote>

control_access_param is the correct name since we are actually converting a ControlAccessParam object to dictionary. AccessControlSetting is a different param.
Rename the param from "control_access" to "control_access_param_xml"/"control_access_param_blob"

</blockquote></details>

removed the control_access_params_to_dict  and currently having access_setting_to_dict and access_control_settings_to_dict. Having these 2 methods is more readable

---

*[pyvcloud/vcd/utils.py, line 269 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0Qrh33yxOQbuPskO6G:-L0RRrhqU_yQVnNBi4c5:b-p3afvt) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/utils.py#L269)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

vcd-cli has a method `as_table` in the utils.py module that is a general utility method to display and array of entities, already transformed to dictionaries. `vapp_to_dict` is another example that flattens out a hierarchical object into key-value pairs
</blockquote></details>

removed the control_access_params_to_dict  and currently having access_setting_to_dict and access_control_settings_to_dict. Having these 2 methods is more readable

---

*[pyvcloud/vcd/utils.py, line 269 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0Qs9rwJ6gaZvvxCrEI:-L0RQqT8RHlfI17VrxNl:bormtg3) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/utils.py#L269)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

None check on control_access?
</blockquote></details>

control_access = self.client.get_linked_resource(
            vdc_resource, RelationType.DOWN,
            EntityType.CONTROL_ACCESS_PARAMS.value)
        return process_access_control_setting(control_access)

if client.get_linked_resource() is successful, the control_accesss is never null. I don't think we need to check for null here.

---

*[pyvcloud/vcd/utils.py, line 271 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0RKA4ZhW_AjQOU0XAg:-L0RVN6OQVy_6CItHKw1:b-896fix) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/utils.py#L271)):*
<details><summary><i>Previously, rocknes wrote…</i></summary><blockquote>

Change the param to access_control_settings
</blockquote></details>

Done.

---

*[pyvcloud/vcd/utils.py, line 277 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0QrsVfFJaU8ECe7P0S:-L0RRTXL9I_ibzOhfdPn:b94ndwp) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/utils.py#L277)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

result = control_access_params_to_dict.. can be moved to first line.
</blockquote></details>

i removed control_access_params_to_dict so it is not applicable now.

---

*[pyvcloud/vcd/vapp.py, line 278 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0QtcO-GLd90jOsffoz:-L0RXHJ8ILEWz34mhWen:b-896fix) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/vapp.py#L278)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

return dict
</blockquote></details>

Done.

---

*[pyvcloud/vcd/vapp.py, line 285 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0Qtf9rJVWMSmP7r9d3:-L0RVlVOeg_cMqUpcoG6:b-896fix) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/vapp.py#L285)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

return access_control_settings_to_dict(access_control) makes more sense.
</blockquote></details>

Done.

---

*[tests/vcd_vapp.py, line 265 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0Qtz1JY8dpJz2dTqq9:-L0RWd3XjLsjcyzfyAQR:b-i2xnyl) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/tests/vcd_vapp.py#L265)):*
<details><summary><i>Previously, sahithi (Sahithi Ayloo) wrote…</i></summary><blockquote>

assert access_control not None? It doesn't return list of records, right?
</blockquote></details>

get_access_control_settings will return an empty dict. We dont need to check for None here

---

*[tests/vcd_vdc.py, line 29 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0QuEkSslqtjjEw-dIj:-L0RWoofDAOm5sLJR7e7:b-hde8pu) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/tests/vcd_vdc.py#L29)):*
get_access_control_settings will return a empty dict. We dont need to check for None here.

>Could it be len(access_control) == 0 and still be a valid vDC configuration?
No

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8)*
<!-- Sent from Reviewable.io -->

rocknes commented:
Reviewed 6 of 6 files at r3.
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---

*[pyvcloud/vcd/org.py, line 644 at r3](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0SN1NpL-ipVzguPsse:-L0SN1NpL-ipVzguPssf:b-drxncz) ([raw file](https://github.com/anusuyar/pyvcloud/blob/5914af7b3c68fa6130a52d2dd855ed6243635dcb/pyvcloud/vcd/org.py#L644)):*
> ```Python
>         """  # NOQA
>         catalog_resource = self.get_catalog(name=catalog_name)
>         access_control_settings = self.client.get_linked_resource(
> ```

If we are retrieving ControlAccessParams I think the variable and function name should reflect that. access_control_settings* is confusing in that case.

---

*[pyvcloud/vcd/utils.py, line 249 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0RKffCefsDdOYvpNum:-L0SNu5GdiFiseHYAhrE:bcjbsxc) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/utils.py#L249)):*
<details><summary><i>Previously, anusuyar wrote…</i></summary><blockquote>

Done.
</blockquote></details>

No, don't send in the extra param n. This looks hacky. Either merge this function with the other function viz. 
access_control_settings_to_dict (which in my opinion should be named control_access_params_to _dict !).  Or else, return the dict without the manipulating the keys (i.e. appending the value of 'n'), and in the other function modify the key while appending the content of this dict to the other result dict.

---

*[pyvcloud/vcd/utils.py, line 259 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0QrR7tBJeWGlFBstTG:-L0SOcrMSa0m9JZHFkKe:b6t0kqt) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/utils.py#L259)):*
<details><summary><i>Previously, anusuyar wrote…</i></summary><blockquote>

removed the control_access_params_to_dict  and currently having access_setting_to_dict and access_control_settings_to_dict. Having these 2 methods is more readable
</blockquote></details>

Don't rename to access_control_settings_to_dict, we are converting a ControlAccessParams object to dict!

---

*[pyvcloud/vcd/vapp.py, line 285 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0Qtf9rJVWMSmP7r9d3:-L0SP8JyVxteuIoye2Wl:b-uabfww) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/vapp.py#L285)):*
<details><summary><i>Previously, anusuyar wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Variable access_control_settings should be control_access_params

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
Reviewed 6 of 6 files at r3.
Review status: all files reviewed at latest revision, 12 unresolved discussions.

---

*[pyvcloud/vcd/org.py, line 644 at r3](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0SN1NpL-ipVzguPsse:-L0VB4IHcNZ9Ce29dYUn:blgdvjf) ([raw file](https://github.com/anusuyar/pyvcloud/blob/5914af7b3c68fa6130a52d2dd855ed6243635dcb/pyvcloud/vcd/org.py#L644)):*
<details><summary><i>Previously, rocknes wrote…</i></summary><blockquote>

If we are retrieving ControlAccessParams I think the variable and function name should reflect that. access_control_settings* is confusing in that case.
</blockquote></details>

FYI in VCDA-335 we are going to decide if we standardize on returning the native XML and leave the client of the method to convert to dict or other representation. We can leave as-is right now to close this ticket, but we might revisit in the future.

---

*[pyvcloud/vcd/utils.py, line 269 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0Qrh33yxOQbuPskO6G:-L0VDKENRM7-RZFJDu0e:b32w0x) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/utils.py#L269)):*
<details><summary><i>Previously, anusuyar wrote…</i></summary><blockquote>

removed the control_access_params_to_dict  and currently having access_setting_to_dict and access_control_settings_to_dict. Having these 2 methods is more readable
</blockquote></details>

OK

---

*[pyvcloud/vcd/utils.py, line 255 at r3](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0VCbAwHw7GP1fbdQh2:-L0VCbAwHw7GP1fbdQh3:b-b6hp9i) ([raw file](https://github.com/anusuyar/pyvcloud/blob/5914af7b3c68fa6130a52d2dd855ed6243635dcb/pyvcloud/vcd/utils.py#L255)):*
> ```Python
> def access_setting_to_dict(access_setting, n):
>     result = {}
>     access_str = 'access-settings'
> ```

I suggested using dash `-` but please use the underscore instead `_` as the token separator. That way the key name can be used directly as a valid variable name in python or key name in yaml if needed.

---

*[tests/vcd_vdc.py, line 29 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0QuEkSslqtjjEw-dIj:-L0VENxYGQrUUw-gdh5T:b32w0x) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/tests/vcd_vdc.py#L29)):*
<details><summary><i>Previously, anusuyar wrote…</i></summary><blockquote>

get_access_control_settings will return a empty dict. We dont need to check for None here.

>Could it be len(access_control) == 0 and still be a valid vDC configuration?
No
</blockquote></details>

OK

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8)*
<!-- Sent from Reviewable.io -->

Commit:
Addressed review comments.

anusuyar commented:
Review status: 5 of 6 files reviewed at latest revision, 12 unresolved discussions.

---

*[pyvcloud/vcd/org.py, line 644 at r3](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0SN1NpL-ipVzguPsse:-L0aRpFtZ-N9Ri6M6tMR:b-tf04au) ([raw file](https://github.com/anusuyar/pyvcloud/blob/5914af7b3c68fa6130a52d2dd855ed6243635dcb/pyvcloud/vcd/org.py#L644)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

FYI in VCDA-335 we are going to decide if we standardize on returning the native XML and leave the client of the method to convert to dict or other representation. We can leave as-is right now to close this ticket, but we might revisit in the future.
</blockquote></details>

I prefer leaving the name as is.  get_acess_control_setting() or access_control_settings is an intuitive name for those who do not know what tags are present in the xml.
Vcloud doc mentions this resourse as 'access control settings'
Use this kind of link to specify new ```access control settings```  for the vApp identified in the href

From https://vdc-repo.vmware.com/vmwb-repository/dcr-public/3ae3f17c-6666-4efa-83bd-3dae5031d559/08a66e37-540e-4987-85b0-ba1cdd40f7c6/vcloud_sp_api_guide_29_0.pdf

```Access Control for vApps

An administrator or vApp owner can control access to a vApp.
Each VApp element includes two types of access control links:
Links where rel="down".
<Link
 rel="down"
 type="application/vnd.vmware.vcloud.controlAccess+xml"
 href="https://vcloud.example.com/api/vApp/vapp-id/controlAccess/"/>
Use this kind of link to retrieve the access control settings for the vApp identified in the href value.

 Links where rel="controlAccess".
<Link
 rel="controlAccess"
 type="application/vnd.vmware.vcloud.controlAccess+xml"
 href="https://vcloud.example.com/api/vApp/vapp-id/action/controlAccess/"/>
Use this kind of link to specify new **access control settings**  for the vApp identified in the href value. You
specify the new access control settings in a ControlAccessParams element that you post to the URL that
the href value of this link specifiesǯ See “Update vApp Access Controls,” on page 127 for an example```
```


---

*[pyvcloud/vcd/utils.py, line 249 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0RKffCefsDdOYvpNum:-L0aSUw4-KTPNCjCf7ep:b-rvsxhg) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/utils.py#L249)):*
<details><summary><i>Previously, rocknes wrote…</i></summary><blockquote>

No, don't send in the extra param n. This looks hacky. Either merge this function with the other function viz. 
access_control_settings_to_dict (which in my opinion should be named control_access_params_to _dict !).  Or else, return the dict without the manipulating the keys (i.e. appending the value of 'n'), and in the other function modify the key while appending the content of this dict to the other result dict.
</blockquote></details>

Done. I will merge the two functions.

---

*[pyvcloud/vcd/utils.py, line 259 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0QrR7tBJeWGlFBstTG:-L0aQau8fo9-XOV3138A:bywrb15) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/utils.py#L259)):*
<details><summary><i>Previously, rocknes wrote…</i></summary><blockquote>

Don't rename to access_control_settings_to_dict, we are converting a ControlAccessParams object to dict!
</blockquote></details>

I think access_control_settings_to_dict is appropriate to be in sync with other similar methods vdc_to_dict , org_to _dict and vapp_to_dict etc. 

---

*[pyvcloud/vcd/utils.py, line 269 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0Qrh33yxOQbuPskO6G:-L0aYla5zeHXYQTXcBdJ:bpw9n0) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/utils.py#L269)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

OK
</blockquote></details>

I have decided to have single function to which will do the conversion to dict

---

*[pyvcloud/vcd/utils.py, line 255 at r3](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0VCbAwHw7GP1fbdQh2:-L0aYxCDkcQgQvV7y1B8:b-896fix) ([raw file](https://github.com/anusuyar/pyvcloud/blob/5914af7b3c68fa6130a52d2dd855ed6243635dcb/pyvcloud/vcd/utils.py#L255)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

I suggested using dash `-` but please use the underscore instead `_` as the token separator. That way the key name can be used directly as a valid variable name in python or key name in yaml if needed.
</blockquote></details>

Done.

---

*[pyvcloud/vcd/vapp.py, line 285 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0Qtf9rJVWMSmP7r9d3:-L0aSnFM0oA2uk00uu3P:b-2j71k9) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/vapp.py#L285)):*
<details><summary><i>Previously, rocknes wrote…</i></summary><blockquote>

Variable access_control_settings should be control_access_params
</blockquote></details>

I dont think we have to exactly translate the entityType. access_control_settings clearly specify what kind of resouse it is. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 12 unresolved discussions.

---

*[pyvcloud/vcd/utils.py, line 255 at r3](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0VCbAwHw7GP1fbdQh2:-L0bZ-8uFYzHwJMpAWe-:bmmwqre) ([raw file](https://github.com/anusuyar/pyvcloud/blob/5914af7b3c68fa6130a52d2dd855ed6243635dcb/pyvcloud/vcd/utils.py#L255)):*
<details><summary><i>Previously, anusuyar wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Replace also the other dashes in the key: `%s-%s-subject-name` should be replaced with `%s_%s_subject_name`.

---

*[pyvcloud/vcd/vapp.py, line 285 at r2](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0Qtf9rJVWMSmP7r9d3:-L0b_Moy5c7GcTAXlayZ:b-guvspu) ([raw file](https://github.com/anusuyar/pyvcloud/blob/19f9209cd367ca438d533d972f38fa51012e0303/pyvcloud/vcd/vapp.py#L285)):*
<details><summary><i>Previously, anusuyar wrote…</i></summary><blockquote>

I dont think we have to exactly translate the entityType. access_control_settings clearly specify what kind of resouse it is. 
</blockquote></details>

Either way could work, but `access_control_settings` might be more descriptive of what it represents. `_params` refers to the API call parameters.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8)*
<!-- Sent from Reviewable.io -->

Commit:
Replacing '-' to '_' in the access_setting string.

anusuyar commented:
Review status: 5 of 6 files reviewed at latest revision, 12 unresolved discussions.

---

*[pyvcloud/vcd/utils.py, line 255 at r3](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0VCbAwHw7GP1fbdQh2:-L0btU0rw4CjQ9bnwvG9:b-896fix) ([raw file](https://github.com/anusuyar/pyvcloud/blob/5914af7b3c68fa6130a52d2dd855ed6243635dcb/pyvcloud/vcd/utils.py#L255)):*
<details><summary><i>Previously, pacogomez (Paco Gómez) wrote…</i></summary><blockquote>

Replace also the other dashes in the key: `%s-%s-subject-name` should be replaced with `%s_%s_subject_name`.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-:-L0bxo7rpLzYm58TNxzi:bnfp4nl)*
<!-- Sent from Reviewable.io -->

rocknes commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[pyvcloud/vcd/utils.py, line 253 at r5](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0fTSP3tvfrpOTGQS2v:-L0fTSP3tvfrpOTGQS2w:b-tkpe61) ([raw file](https://github.com/anusuyar/pyvcloud/blob/0d526e07abf4063e06160eabe51987f6e23b6e3e/pyvcloud/vcd/utils.py#L253)):*
> ```Python
> 
> 
> def access_control_settings_to_dict(access_control_settings):
> ```

Can you please add a line of documentation stating that the param is a ControlAccessParamType xml object. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-:-L0fTyLh7E0LcqEWFVtP:bnfp4nl)*
<!-- Sent from Reviewable.io -->

pacogomez commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 8 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-:-L0fdP-jY6OFw7v5AlVC:bnfp4nl)*
<!-- Sent from Reviewable.io -->

Commit:
Add docstring to access_control_settings_to_dict

sahithi commented:
Reviewed 5 of 6 files at r3, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8)*
<!-- Sent from Reviewable.io -->

anusuyar commented:
Review status: 5 of 6 files reviewed at latest revision, 3 unresolved discussions.

---

*[pyvcloud/vcd/utils.py, line 253 at r5](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8#-L0fTSP3tvfrpOTGQS2v:-L0fZo_jdsplbXNsHwcd:b-896fix) ([raw file](https://github.com/anusuyar/pyvcloud/blob/0d526e07abf4063e06160eabe51987f6e23b6e3e/pyvcloud/vcd/utils.py#L253)):*
<details><summary><i>Previously, rocknes wrote…</i></summary><blockquote>

Can you please add a line of documentation stating that the param is a ControlAccessParamType xml object. 
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8)*
<!-- Sent from Reviewable.io -->

rocknes commented:
Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/anusuyar/pyvcloud/8)*
<!-- Sent from Reviewable.io -->

Commit:
Merge changes from the vmware/master
